mender-auth/cli/keystore.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-auth/cli/keystore.hpp>
+
+#include <common/log.hpp>
+#include <common/crypto.hpp>
+
+
+namespace mender {
+namespace auth {
+namespace cli {
+
+using namespace std;
+
+namespace log = mender::common::log;
+
+namespace crypto = mender::common::crypto;
+
+class KeyStoreErrorCategoryClass : public std::error_category {
+public:
+	const char *name() const noexcept override;
+	string message(int code) const override;
+};
+
+enum ErrorCode {
+	NoError = 0,
+	NoKeysError,
+	StaticKeyError,
+};
+
+const KeyStoreErrorCategoryClass KeyStoreErrorCategory;
+
+const char *KeyStoreErrorCategoryClass::name() const noexcept {
+	return "KeyStoreErrorCategory";
+}
+
+string KeyStoreErrorCategoryClass::message(int code) const {
+	switch (code) {
+	case NoError:
+		return "Success";
+	case NoKeysError:
+		return "No keys found";
```

Comment:
This message is a bit misleading as to what it's used for. What it means is really "No key in memory", isn't it? I had to go back and forth a few times because I thought it was referring to missing keys on the filesystem.

Reply:
It means that no key has been loaded or generated. I'll rephrase. I took it from the [legacy client](https://github.com/mendersoftware/mender/blob/1a86f02ab83795b136f7d0bc08f5f39701ea3b21/store/keystore.go#L32).